src/endpoint.rs

```diff
@@ -30,25 +28,25 @@ macro_rules! endpoint {
             rustemon_client.get_with_limit_and_offset::<NamedApiResourceList<$type>>($name, limit, offset).await
         }
 
-        /// Returns all pages from the given resource.
+        /// Returns all entries from the given resource.
         ///
         /// # Arguments
         ///
         /// `rustemon_client` - The [RustemonClient] to use to access the resource.
-        pub async fn get_all_pages(rustemon_client: &RustemonClient) -> Result<Vec<NamedApiResource<$type>>, Error> {
-            let mut page = get_page(rustemon_client).await?;
-            let mut all_pages = Vec::with_capacity(page.count as usize);
+        pub async fn get_all_entries(rustemon_client: &RustemonClient) -> Result<Vec<NamedApiResource<$type>>, Error> {
+            let mut first_page = get_page(rustemon_client).await?;
+            let first_page_entries_count = first_page.results.len() as i64;
 
-            all_pages.append(&mut page.results);
+            let total_entries_count = first_page.count;
+            let other_entries_count = total_entries_count - first_page_entries_count;
 
-            while let Some(ref next_page_url) = page.next {
-                let next_page_url = Url::parse(next_page_url).unwrap();
-                page = rustemon_client.get_by_url::<NamedApiResourceList<$type>>(next_page_url).await?;
+            let mut all_entries = Vec::with_capacity(total_entries_count as usize);
+            all_entries.append(&mut first_page.results);
 
-                all_pages.append(&mut page.results);
-            }
+            let mut other_entries = get_page_with_param(first_page_entries_count, other_entries_count, &rustemon_client).await?;
```

Comment:
Can you please remove the `&` before rustemon_client please ? 
I believe it's not necessary